test/pet.test.js

```diff
@@ -0,0 +1,189 @@
+const Pet = require('../src/pet');
+
+let testPet;
+
+const deadMessage = "Your pet is no longer alive :("
+
+beforeEach(() => {
+    testPet = new Pet();
+  });
+
+describe('constructor', () => {
+    it('should return a new pet with a name', () => {
+        expect(new Pet()).toBeInstanceOf(Object);
+        expect(new Pet().name).toBe("An unloved pet");
+        expect(new Pet('Manny').name).toBe("Manny");
+    })
+
+    it('should return a new pet with an age', () => {
+        expect(new Pet().age).toBe(0);
+    })
+
+    it('shold return a new pet with a hunger', () => {
+        expect(new Pet().hunger).toBe(0);
+    })
+
+    it('shold return a new pet with a fitness', () => {
+        expect(new Pet().fitness).toBe(10);
+    })
+})
+
+describe('growUp', () => {
+    it('should make the pet grow', () => {
+        testPet.growUp();
+        expect(testPet.age).toBe(1);
+    })
+
+    it('should make the pet hunger increase', () => {
+        testPet.growUp();
+        expect(testPet.hunger).toBe(5);
+    })
+
+    it('should make the pet fitness decrease', () => {
+        testPet.growUp();
+        expect(testPet.fitness).toBe(7);
+    })
+
+    it('should throw an exception with "Your pet is no longer alive :(", if dead', () => {
+        testPet.age = 30;
+        expect(() => testPet.growUp()).toThrow(deadMessage);
+    })
+})
+
+describe('walk', () => {
+    it('should increase fitness by 4', () => {
+        testPet.fitness = 5;
+        testPet.walk();
+        expect(testPet.fitness).toBe(9);
+    })
+
+    it('should have a max fitness of 10', () => {
+        testPet.walk();
+        expect(testPet.fitness).toBe(10);
+        testPet.fitness = 8;
+        testPet.walk();
+        expect(testPet.fitness).toBe(10);
+        testPet.fitness = 4;
+        testPet.walk();
+        expect(testPet.fitness).toBe(8);
+    })
+
+    it('should throw an exception with "Your pet is no longer alive :(", if dead', () => {
+        testPet.age = 30;
+        expect(() => testPet.walk()).toThrow(deadMessage);
+    })
+})
+
+describe('feed', () => {
+    it('should decrease hunger by 3', () => {
+        testPet.growUp();
```

Comment:
When you use another function in your tests, you are assuming that function must be working correctly, when it may not. 
Instead of using growUp here to increase the hunger, define it first and then other functions won't mess with your test for feed

Reply:
Very good point! :D